### PR TITLE
Feature add gmail auth

### PR DESCRIPTION
**Pull request by litcodeur**

### CODE REVIEW

File: client/src/pages/gmailAuth/AuthRoute.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import axios from 'axios'
+import queryString from 'query-string'
+import withAuth from "common/withAuth";
+import useSnackbar from 'common/useSnackbar';
+
+// A Route that responds when OAuth redirects the user back to our application
+
+const Auth = (props) => {
+
+    const showSnackBar = useSnackbar()
```
Comment (skassam21) [inline client/src/pages/gmailAuth/AuthRoute.js:11]: Are you using prettier on the front-end - noticed things are missing like ; for example

File: client/src/pages/gmailAuth/AuthRoute.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            } catch (error) {
+                showSnackBar('Failed to Authenticate')
+            }
+        }
+        checkAuth()
+    }, [props.history, props.location.search, showSnackBar])
+
+    return (
+        <div>
+            {props.history.push("/")}
```
Comment (skassam21) [inline client/src/pages/gmailAuth/AuthRoute.js:31]: I don't know if I love this flow, I think a success dialog on top of the current content will be better, but we can do that later

File: server/src/controllers/gmail.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
+const { google } = require('googleapis')
+const DataBaseConnectionError = require('../errors/database-connection-error')
+
+// The controller for Gmail Client
+class Gmail {
+
+    // Get an OAuthClient with configurations
+    static get_oAuthClient = () => {
```
Comment (skassam21) [inline server/src/controllers/gmail.js:8]: make this camelCase

File: server/src/controllers/gmail.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        const oAuth2Client = new google.auth.OAuth2(
+            client_id, client_secret, redirect_uris);
+
+        return oAuth2Client
+    }
+
+    // Responds to route that gets the Authroriztion URL from google
+    static getAuthURL = async (req, res) => {
+
+        const SCOPES = ['https://www.googleapis.com/auth/gmail.readonly'];
```
Comment (skassam21) [inline server/src/controllers/gmail.js:22]: we will need also the ability to send emails

File: server/src/controllers/gmail.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+                return res.status(400).send({
+                    "errors": [{
+                        "message": "Error Signing In"
+                    }]
+                })
+            }
+
+            req.session = {
+                ...req.session,
+                token: token
```
Comment (skassam21) [inline server/src/controllers/gmail.js:54]: Got it - yeah we are going to need to store this in the database like we discussed

**Review by skassam21 — COMMENTED** [review]
Looks overall good, we just discuss the improvements we will need in our standup

File: server/src/controllers/gmail.js (hunk 4 above)
Comment (litcodeur) [reply server/src/controllers/gmail.js:22]: Okay will add it in next pr

**Pull request closed without merging**